src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/LeaverSteps.java

```diff
@@ -0,0 +1,252 @@
+
+  @Value("${service.rabbit.url}")
+  private String rabbitLocation;
+
+  @Value("${service.rabbit.username}")
+  private String rabbitUsername;
+
+  @Value("${service.rabbit.password}")
+  private String rabbitPassword;
+
+  @Value("${addeco.baseUrl}")
+  private String mockAdeccoUrl;
+  private String adeccoWorker;
+
+  @Value("${rcaExtractLocation}")
+  private String rcaExtractLocation;
+
+  @Before
+  public void setup() throws Exception {
+    adeccoMockUtils.clearMock();
+    adeccoMockUtils.cleardb();
+
+    adeccoMockUtils.enableRequestRecorder();
+
+    gatewayEventMonitor = new GatewayEventMonitor();
+    gatewayEventMonitor.enableEventMonitor(rabbitLocation, rabbitUsername, rabbitPassword);
+  }
+
+  @After
+  public void tearDownGatewayEventMonitor() throws IOException {
+    adeccoMockUtils.disableRequestRecorder();
+    gsuiteMockUtils.clearMock();
+    snowMockUtils.clearMock();
+    xmaMockUtils.clearMock();
+    gatewayEventMonitor.tearDownGatewayEventMonitor();
+  }
+
+  @Given("An employee exists in {string} with an id of {string}")
+  public void we_recieve_an_employee_with_an_id_of(String source, String id) {
+
+    adeccoResponse = AdeccoPeopleFactory.buildFransicoBuyo(id);
+    adeccoResponse.setContractStartDate(LocalDate.now().toString());
+  }
+
+  @Given("an assignment status of {string}")
+  public void an_assignment_status_of(String assignmentStatus) {
+    adeccoResponse.setStatus(assignmentStatus);
+  }
+
+  @Given("a closing report status of {string}")
+  public void a_closing_report_status_of(String crStatus) {
+    adeccoResponse.setCrStatus(crStatus);
+  }
+
+  @Given("a role id of {string}")
+  public void a_role_id_of(String roleId) {
+    adeccoResponse.getResponseJob().setRoleId(roleId);
+  }
+
+  @Given("an operational end date of {string}")
+  public void an_operational_end_date_of(String date) {
+    adeccoResponse.setOperationalEndDate(date);
+  }
+
+  @When("we receive a job role update from adecco for employee  {string}")
+  public void we_receive_a_job_role_update_from_adecco_for_employee(String id) {
+    adeccoResponse.setContractStartDate(LocalDate.now().toString());
+  }
+
+  @Given("we ingest them")
+  public void we_ingest_them() throws IOException {
+    List<AdeccoResponse> adeccoResponseList = new ArrayList<>();
+    adeccoResponseList.add(adeccoResponse);
+    adeccoMockUtils.addUsersAdecco(adeccoResponseList);
+
+    fsdrUtils.ingestAdecco();
+    fsdrUtils.ingestRunFSDRProcess();
+
+  }
+
+  @When("the employee is sent to all downstream services")
+  public void theEmployeeIsSentToAllDownstreamServices() throws Exception {
+
+    fsdrUtils.ingestGsuit();
+    fsdrUtils.ingestXma();
+    fsdrUtils.ingestSnow();
+    fsdrUtils.ingestGranby();
+    fsdrUtils.lwsExtract();
+    fsdrUtils.rcaExtract();
+  }
+
+  @When("the employee is sent to LWS")
+  public void theEmployeeIsSentToLWS() throws Exception {
+
+    fsdrUtils.lwsExtract();
```

Comment:
wait for lws_complete event?
Things run slower on the cloud, and steve had no end of issues with race conditions.